examples/scale/scenarios/basic_scenario.go

```diff
@@ -0,0 +1,23 @@
+package scenarios
+
+import (
+	"fmt"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+// BasicScenario is a struct with Scenario embedded that implements mmf.Run and evaluator.Evaluate methods.
+type BasicScenario struct {
```

Comment:
I think we should be exposing an interface or single scenario type from this package?

Reply:
More detail:
We may implement a handful of different kinds of scenarios, and they should all be hidden inside the package implementation.

Reply:
I tried to expose an interface but golang interface doesn't support data fields in it - The only way to make interface works is either by implementing getters for each of the controllable parameters which looks a bit verbose to me.

Exposing a single scenario type also doesn't fit to our use case since `pb.RegisterXXX` handler requires a concrete implementation of an interface.

There is a way to "implement" a abstract class in go but it's pretty tricky so I chose to comprise with embedding.

Reply:
The function exposed here doesn't have to actually implement the grpc service.

I think the best option is to return a struct with all the values on it.  A "basicScenario" function could return this Scenario struct with all of the fields populated.

Reply:
Also, there's a clever trick when you have a function but want an interface:

```
type interface Fooer {
  foo(int) bool
}

func myfunc(a int) bool {
  return a == 0
}

type FooerFunc func(int) bool

func (f FooerFunc) Foo(a int) bool {
  return f(a)
}

// Can now use FooerFunc(myfunc) wherever you need a Fooer.

```

Reply:
Ah this is a neat trick, thanks for the suggestion.